### PR TITLE
We now are having a global items page

### PR DESCRIPTION
**Pull request by annando**
The current situation was that the community page was configurable between showing public posts from local users or all known global (public) posts.

Now we do have two separated pages.

Additionally there are some preparations for being able to comment on these pages in the future.

### CODE REVIEW

File: src/Module/Global.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * @file src/Module/Global.php
```
Comment (MrPetovan) [inline src/Module/Global.php]: This file should be called "GlobalModule" as the class it holds is called.

File: src/Module/Global.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+use dba;
+
+/**
+ * Global module
+ *
+ * Displays global posts on the server
+ *
+ * @author heluecht@pirati.ca
+ */
+class GlobalModule extends BaseModule {
```
Comment (MrPetovan) [inline src/Module/Global.php]: Standards: Please add a new line after class declarations.

File: src/Module/Global.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	public static function init() {
```
Comment (MrPetovan) [inline src/Module/Global.php]: Standards: Please add a new line after function declarations.

File: src/Module/Global.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		if (!local_user()) {
+			unset($_SESSION['theme']);
+			unset($_SESSION['mobile-theme']);
+		}
+	}
+
+	public static function content($update = 0) {
```
Comment (MrPetovan) [inline src/Module/Global.php]: Standards: Please add a new line after function declarations.

File: src/Module/Global.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+		$t = get_markup_template("community.tpl");
+		return replace_macros($t, array(
+			'$content' => $o,
+			'$header' => t("Global Timeline"),
+			'$show_global_community_hint' => Config::get('system', 'show_global_community_hint'),
+			'$global_community_hint' => t("This community stream shows all public posts received by this node. They may not reflect the opinions of this node’s users.")
+		));
+	}
+
+	private static function getPublicItems($start, $itemspage) {
```
Comment (MrPetovan) [inline src/Module/Global.php]: Standards: Please add a new line after function declarations.

**Comment by tobiasd** [comment]
On Wed, 03 Jan 2018 13:53:48 +0000 (UTC)
Michael Vogel <notifications@github.com> wrote:

> The current situation was that the community page was configurable
> between showing public posts from local users or all known global
> (public) posts.

What's the reasoning behind having two public streams?

**Comment by annando** [comment]
The two are different, since the first one shows what people on the server are writing and the second one shows what the network is posting.

Formerly an admin had to chose which stream to show. Having it now under two different pages we do it like GNU Social or Mastodon are doing.

**Comment by MrPetovan** [comment]
So:
- the `global` module is showing the network posts.
- the `network` module is showing the timeline posts

I feel like it's a renaming just waiting to happen.

**Comment by tobiasd** [comment]
> The two are different, since the first one shows what people on the
> server are writing and the second one shows what the network is
> posting.

I hope one can still disable them both.

**Comment by rabuzarus** [comment]
Some notes according to GUI/UX
Wouldn't it be better to to have one `community` module with entries in the tab bar with `global` and `local` to switch between both?

**Comment by annando** [comment]
@rabuzarus From an UX position I don't see a benefit to have two instead of a single click to reach the content.

@tobiasd Nearly as before you can disable this for visitors to your system. But I made it always available for local users since this will help especially new users to find content.

**Comment by annando** [comment]
@MrPetovan ```global``` shows global posts, ```network``` shows posts from your personal network.

**Comment by MrPetovan** [comment]
I know your reasoning, but I feel like it isn't the most popular terminology and as such it can be confusing for new users, on top of `global` being a reserved word in PHP.

**Comment by rabuzarus** [comment]
@annando 
1.) As GUI designer I see problems with the space - another icon in the topbar - there could be problems with smaller screens - we have also no possibility to put this icon in the topbar on mobile devices
2.) With regard to UX: We should try to group similar Functions - global/local community are similar and differ only in their range. If you want you can see the local community page as a filter of the the global community page

**Comment by MrPetovan** [comment]
I agree with @rabuzarus about using the dropdown, even if it can be a separate (renamed) module.

**Comment by hoergen** [comment]
Instead of using the name "global" we could name it "fediverse"?

**Comment by MrPetovan** [comment]
It's another idea, however I still stand with `timeline` which should also be the default display when logged in for the current `network` module, and `network` for the proposed `global` module.

- https://friendi.ca/ (anonymous) => `community`/`login` module depending on settings
- https://friendi.ca/ (logged in) => `timeline` module
- https://friendi.ca/timeline => `timeline` module
- https://friendi.ca/community => `community` module
- https://friendi.ca/network => `global` module

**Comment by AlfredSK** [comment]
Another possibility would be to do it like Mastodon. They have a local timeline and a federated timeline.
But I also like @MrPetovan idea.

**Comment by MrPetovan** [comment]
@AlfredSK That's what we're talking about. We currently have a user timeline called `network`, a local timeline called `community`, and this new federated timeline is called `global` for now, which isn't ideal since it's a reserved PHP word.

**Comment by tobiasd** [comment]
> Instead of using the name "global" we could name it "fediverse"?

Why Fediverse and not Federation? Why not something that reflects the entity better? "Whole Known Network" or "WKN" is somethign I read before.

**Comment by tobiasd** [comment]
> @tobiasd Nearly as before you can disable this for visitors to your
> system. But I made it always available for local users since this
> will help especially new users to find content.

After thinking it over a bit, I still don't see a need for this. For me one community timeline, which the node admin can disable if not needed, is enough.

**Comment by annando** [comment]
Now I'm slightly confused. Should we make it accessible under two different paths or under a single path with a submenu like the network page? Should we rename all path names or what should we do?

I, personally, would like to delay the decision to later, so that we can discuss about it. For me, this PR here is only a preparation for some more work to support commenting on these timelines. Means: I would like to focus on the backend and would like to delegate the frontend work.

**Comment by AlfredSK** [comment]
Maybe we should discuss this in the support forum to get more input and opinions from the community.
Issue 1: How should we present the two community streams in the GUI? Two top level sections or one top level section with two sub-sections?
Issue 2: How should we name them?

**Comment by annando** [comment]
Okay, so now we do have a single page.

**Comment by annando** [comment]
From my point of view I'm now done with this PR. When ```$a->config['system']['comment_public'] = true;``` is activated in .htconfig.php, we can now act on the community page like on the network page. Means: we can comment and like and the content is automatically updated when we are on the top of the page.

**Comment by AlfredSK** [comment]
It's working. :-) Commenting and liking is not possible for Friendica postings, right?

**Comment by zeroadam** [comment]
I see the two links on /community as well. I assume there is not an admin option yet to enable `$a->config['system']['comment_public'] = true;`

**Comment by annando** [comment]
The config value "comment_public" is just for the testing phase, until the development is done.

@AlfredSK I hope to develop some workaround to do this for Friendica postings as well. The quick way (I have to check if this works) would be to transmit it via the Diaspora protocol. The other way is much more complicated, since our protocol is not meant to allow this.

**Comment by MrPetovan** [comment]
Related to #4117

**Comment by MrPetovan** [comment]
I would like to point out that a community page still doesn't make sens on a single user instance, however the tab still is available.

**Comment by annando** [comment]
The global page does make much sense, since you can subscribe to the relay service which would you provide with additional content.

**Comment by MrPetovan** [comment]
Yes, the global tab makes sense, but not the local tab.

**Comment by annando** [comment]
Okay, see my PR.

**Pull request merged**